max macro - compiler errors

My source file has 4 lines:
````
#include <user_config.h>
#include <SmingCore/SmingCore.h>
#include <Libraries/OneWire/OneWire.h>
#include <algorithm>
````
It doesnt build.

Errors are:

In file included from /mnt/c/opt/esp-open-sdk/xtensa-lx106-elf/xtensa-lx106-elf/include/c++/4.8.5/bits/stl_algo.h:60:0,
                 from /mnt/c/opt/esp-open-sdk/xtensa-lx106-elf/xtensa-lx106-elf/include/c++/4.8.5/algorithm:62,
                 from app/temperatureDS18B20.cpp:12:
/mnt/c/opt/esp-open-sdk/xtensa-lx106-elf/xtensa-lx106-elf/include/c++/4.8.5/bits/algorithmfwd.h:360:41: error: macro "max" passed 3 arguments, but takes just 2
     max(const _Tp&, const _Tp&, _Compare);

--- a/Sming/SmingCore/ArduinoCompat.h
+++ b/Sming/SmingCore/ArduinoCompat.h
@@ -20,12 +20,6 @@ extern "C" {
 #endif
 
 #define abs(x) ((x) > 0 ? (x) : -(x))
-#ifndef min
-#define min(a, b) ((a) < (b) ? (a) : (b))
-#endif
-#ifndef max
-#define max(a, b) ((a) > (b) ? (a) : (b))
-#endif
 #define round(x) ((x) >= 0 ? (long)((x) + 0.5) : (long)((x)-0.5))
 
 void yield();

--- a/Sming/Wiring/WiringFrameworkDependencies.h
+++ b/Sming/Wiring/WiringFrameworkDependencies.h
@@ -22,4 +22,13 @@
 #include "FakePgmSpace.h"
 #include "../SmingCore/pins_arduino.h"
 
+// std::min, etc.
+#include <algorithm>
+
+using std::isinf;
+using std::isnan;
+using std::max;
+using std::min;
+
+
 #endif /* WIRING_WIRINGFRAMEWORKDEPENDENCIES_H_ */

